### PR TITLE
add playbook for karrot.world and fix behavior on bare debian

### PR DESCRIPTION
**Pull request by tiltec**
I copied it from the foodsharing taiwan playbook, adjusted secrets and added the backup script

I didn't run it yet, was afraid of having to fix a broken production deployment.

The site user needs access to the configured offsite location, for now by manually copying the private key from the deploy user. I wonder if I should add that private key to the vault?

The site user also needs access to progres to create arbitrary databases and delete them later on. It would probably be better to have only one backup test db. Maybe this can also be added to the playbook...

### CODE REVIEW

**Comment by nicksellen** [comment]
> I didn't run it yet, was afraid of having to fix a broken production deployment.

We could co-ordinate to be online at the same time, at a reasonably out-of-hours type time, and fix any crap together. Or wait till we're together in person.

> The site user needs access to the configured offsite location, for now by manually copying the private key from the deploy user. I wonder if I should add that private key to the vault?

The ansible user module can generate an ssh key ---> http://docs.ansible.com/ansible/latest/user_module.html

So we could use that, the rsync to offsite location would fail until someone added the public key to it, but thats seems fine. It's a oneoff.

> The site user also needs access to progres to create arbitrary databases and delete them later on. It would probably be better to have only one backup test db. Maybe this can also be added to the playbook...

Would seem nice just to grant access to the site user (or even a seperate site-specific backup user) for just that backup database. Easy to automate it all with ansible :)

**Comment by tiltec** [comment]
Right now, the playbook depends on a GPG key to exist to encrypt the backups. I was looking for ways to manage this:
- perhaps pregenerate the key and store it in vault, then import to system
- use https://github.com/juju4/ansible-gpgkey_generate or similar to generate a key on the fly

I like the first option better, then all people who know the vault password can also decrypt the backups.

**Comment by nicksellen** [comment]
Yup, 1 sounds good to me.

File: playbooks/karrot-world/setup.playbook.yml [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+        workers: 4
+
+    - name: karrot-backups
+      include_role:
+        name: karrot-backups
+      vars:
+        postgresql_database: fstool
+        gpg_id: B88A03E5
+        offsite_server: tiltec@81.30.158.213:backup/
+        gpg_secret_key: !vault |
```
Comment (tiltec) [inline playbooks/karrot-world/setup.playbook.yml]: I couldn't think of a better way than to include the secret key inline (also not sure if it works).

**Pull request merged**